queue.c

```diff
@@ -268,8 +268,13 @@ BaseType_t xQueueGenericReset( QueueHandle_t xQueue,
 
     configASSERT( pxQueue );
 
-    taskENTER_CRITICAL();
+    if( ( pxQueue != NULL ) &&
+        ( pxQueue->uxLength >= 1 ) &&
+        /* Check for multiplication overflow. */
+        ( ( SIZE_MAX / pxQueue->uxLength ) >= pxQueue->uxItemSize ) )

```

Comment:
uxLength and uxItemSize are both UBaseType_t, rather than size_t, so this assumes sizeof( UBaseType_t ) == sizeof( size_t ) -> can we create a local constant that is equivalent to UBaseType_t_MAX?

Reply:
I think SIZE_MAX or PTRDIFF_MAX are reasonable choices.  Given that the `pxQueue->uxLength` and `pxQueue->uxItemSize` are used to do pointer arithmetic and also with `pvPortMalloc()`, we want the product of these to always be  <= to the largest theoretical object.